llm.txt

```diff
@@ -0,0 +1,775 @@
+
+**Problem**: Adding new users in CI/CD environments (like Spacelift) fails because passwords are required but shouldn't be stored in configuration.
+
+**Error**: `Error: password is required for new users but cannot be provided in automated workflows`
+
+```
+Solution: Use generated passwords with temporary access:
+
+# Option 1: Generate random password (recommended for CI/CD)
+resource "random_password" "user_temp_password" {
+  for_each = var.new_users_without_passwords
+  length   = 16
+  special  = true
+}
+
+locals {
+  users_with_generated_passwords = {
+    for email, user in var.users : email => merge(user, {
+      password = try(user.password, random_password.user_temp_password[email].result)
+      change_password_at_next_login = try(user.change_password_at_next_login, true)
+    })
+  }
+}
+
+module "workspace_users_groups" {
+  source = "masterpointio/users-groups-automation/googleworkspace"
+  users  = local.users_with_generated_passwords
+  # ... other configuration
+}
+
+# Output temporary passwords securely (for initial access)
+output "temporary_passwords" {
+  value = {
+    for email, password in random_password.user_temp_password : 
+    email => password.result
+  }
+  sensitive = true
+}
+
+# Option 2: Use consistent temporary password
+variable "temp_password_for_new_users" {
+  description = "Temporary password for new users (they must change on first login)"
+  type        = string
+  sensitive   = true
+  default     = "ChangeMe123!"
+}
+
+# Apply to all new users
+users = {
+  "newuser@company.com" = {
+    primary_email = "newuser@company.com"
+    given_name    = "New"
+    family_name   = "User"
+    password      = var.temp_password_for_new_users
+    change_password_at_next_login = true  # Force password change
+  }
+}
+```
+
+**Workflow Recommendations**:
+1. Use generated passwords for new users in automated environments
+2. Always set `change_password_at_next_login = true` for new users
+3. Communicate temporary passwords through secure channels (not in logs)
+4. Consider using Google's "Send password info to personal email" option manually after Terraform creates the user
+
+### Debug Commands
+
+```bash
+# Check provider authentication
+terraform plan -target="data.googleworkspace_domain_alias.test" 
+
+# Validate configuration
+terraform validate
+
+# See detailed error messages
+TF_LOG=DEBUG terraform apply
+
+# Check specific resource state
+terraform state show 'module.workspace_users_groups.googleworkspace_group.defaults["group_key"]'
+```
+
+## Security Best Practices
+
+### Credential Management
+```bash
+# Use environment variables for sensitive values
+export TF_VAR_google_credentials_file="/secure/path/to/key.json"
+export TF_VAR_super_admin_email="admin@company.com"
+export TF_VAR_user_default_password="SecurePassword123!"
+
+# Never commit credentials to version control
+echo "*.json" >> .gitignore
+echo "terraform.tfvars" >> .gitignore
+```
+
+### Service Account Security
```

Comment:
@westonplatter I feel like this is out of scope for the TF module. What do you think?